### PR TITLE
feat: add `windows`

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: packages/iterator-utilities/src/lib/windows.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * @example
+ * ```typescript
+ * import { windows } from '@sapphire/iterator-utilities';
+ *
+ * const iterable = [1, 2, 3, 4, 5];
+ * console.log([...windows(iterable, 2)]);
+ * // Output: [[1, 2], [2, 3], [3, 4], [4, 5]]
+ * ```
+ */
+export function windows<const ElementType>(iterable: IterableResolvable<ElementType>, count: number): IterableIterator<ElementType[]> {
```
Comment (vladfrangu) [inline packages/iterator-utilities/src/lib/windows.ts:22]: Funny pun but shouldn't this be called `window` not `windows`?

Comment (kyranet) [reply packages/iterator-utilities/src/lib/windows.ts:22]: No, it's a method that makes windows, not a single window.

Also the name is based on the proposal: https://github.com/tc39/proposal-iterator-chunking

File: packages/iterator-utilities/tests/windows.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	});
+
+	test('GIVEN empty iterable THEN returns empty iterable', () => {
+		const iterable: number[] = [];
+		const limit = 5;
+		const result = [...windows(iterable, limit)];
+		expect(result).toEqual([]);
+	});
+
+	test('GIVEN iterable and limit equal to 0 THEN returns empty iterable', () => {
```
Comment (vladfrangu) [inline packages/iterator-utilities/tests/windows.test.ts:35]: wrong test msg

**Review by vladfrangu — DISMISSED** [review]

**Review by favna — APPROVED** [review]
Where are Linux and MacOS?

**Pull request merged**